packages/legend-application-query/src/components/QueryBuilderExplorerPanel.tsx

```diff
@@ -113,6 +116,46 @@ const checkForDeprecatedNode = (
   return false;
 };
 
+const isNodeAlreadyUsed = (
+  node: QueryBuilderExplorerTreeNodeData,
+  projectionState: QueryBuilderProjectionState,
+): boolean => {
+  if (node.id === '@dummy_rootNode' && projectionState.columns.length) {
+    return true;
+  } else if (
+    node instanceof QueryBuilderExplorerTreePropertyNodeData &&
+    projectionState.columns.some(
+      (column) =>
+        column instanceof QueryBuilderSimpleProjectionColumnState &&
+        column.propertyExpressionState.path.startsWith(node.id),
+    )
+  ) {
+    return true;
+  }
+  // For subType
+  else if (node.id.startsWith(TYPE_CAST_TOKEN)) {
+    let propName = '';
+    node.id
+      .split('.')
+      .slice(1)
+      .forEach((p) => {
+        propName = `${propName + p}.`;
```

Comment:
Another approach is to generate the property chain name for this node e.g. (@Firm)/Employee Size

1. buildPropertyExpressionFromExplorerTreeNodeData
2. getPropertyChainName
3. compare column's title

Reply:
@akphi @MauricioUyaguari Which one do you guys prefer?